src/ast/passes/codegen_llvm.cpp

```diff
@@ -771,6 +771,51 @@ void CodegenLLVM::visit(Call &call)
       }
     }
     expr_ = nullptr;
+  } else if (call.func == "has_key") {
+    auto &arg = *call.vargs.at(0);
+    auto &map = static_cast<Map &>(arg);
+    auto [key, scoped_key_deleter] = getMapKey(map, call.vargs.at(1));
+
+    CallInst *lookup = b_.CreateMapLookup(map, key);
+
+    AllocaInst *has_key = b_.CreateAllocaBPF(b_.getInt8Ty(), "has_key");
```

Comment:
CreateBool() in types.cpp defines a boolean as a 1 bit integer. This should probably be a `b_.getInt1Ty()` right?

Reply:
Also, use a phi node here instead of a stack temporary. See https://github.com/bpftrace/bpftrace/commit/caed2944717f985d6295a84e3ff166bc7038e76b for example

Reply:
Ok, I tried. Let me know if I did this right.